saleor/plugins/invoicing/utils.py

```diff
@@ -14,24 +14,31 @@
 MAX_PRODUCTS_PER_PAGE = 13
 
 
-def make_full_invoice_number(number=1):
-    month_and_year = datetime.now().strftime("%m/%Y")
-    return f"{number}/{month_and_year}"
+def make_full_invoice_number(number=None, month=None, year=None):
+    now = datetime.now()
+    current_month = int(now.strftime("%m"))
+    current_year = int(now.strftime("%Y"))
+    month_and_year = now.strftime("%m/%Y")
 
+    if month == current_month and year == current_year:
+        new_number = (number or 0) + 1
+        return f"{new_number}/{month_and_year}"
+    return f"1/{month_and_year}"
 
-def parse_invoice_number(invoice):
-    number = re.match(r"^(\d+)\/", invoice.number).group(1)
-    return int(number)
+
+def parse_invoice_dates(invoice):
+    match = re.match(r"^(\d+)\/(\d+)\/(\d+)", invoice.number)
+    return int(match.group(1)), int(match.group(2)), int(match.group(3))
 
 
 def generate_invoice_number():
-    last_invoice = Invoice.objects.last()
+    last_invoice = Invoice.objects.filter(number__isnull=False).last()
```

Comment:
Why and when we may have invoices without a number?

Reply:
Point of the invoice plugin is to provide it with empty invoice object for manipulation. That object is already created but doesn't have a number yet. This is why we want to fetch the last invoice that did had the number. When the plugin finish its job (this `generate_invoice_number` is actually called in plugin), the number will be there.

Reply:
👍